### PR TITLE
Persistence add standalone

### PR DESCRIPTION
**Pull request by MendelMonteiro**
Add a stand-alone implementation of the Persistence storage so that Zebus with a persistent store can be tested without a full Cassandra cluster.

Currently a simple RocksDb based implementation has been provided which uses the .NET binding [rocksdb-net](https://github.com/warrenfalk/rocksdb-sharp).  An option has been included in the Persistence Runner to launch with the RocksDb storage implementation.

I took this opportunity to refactor the Cassandra storage class as there was a complicated system in place for determining the last updated peers and also an edge case where non acked counts data could be lost.

### CODE REVIEW

File: src/Abc.Zebus.Persistence.Runner/Abc.Zebus.Persistence.Runner.csproj [hunk 1]
```diff
@@ -4,14 +4,27 @@
...
     <ProjectReference Include="..\Abc.Zebus.Persistence.CQL\Abc.Zebus.Persistence.CQL.csproj" />
+    <ProjectReference Include="..\Abc.Zebus.Persistence.RocksDb\Abc.Zebus.Persistence.RocksDb.csproj" />
   </ItemGroup>
 
   <ItemGroup>
     <PackageReference Include="FluentDateTime" Version="1.14.0" />
   </ItemGroup>
 
+  <ItemGroup>
+    <None Update="Abc.Zebus.libzmq.x86.dll">
```
[the remark this answers is not in the document]
Comment (MendelMonteiro) [reply src/Abc.Zebus.Persistence.Runner/Abc.Zebus.Persistence.Runner.csproj]: Indeed

**Comment by MendelMonteiro** [comment]
How were you coming along with moving to Azure DevOps?

**Comment by ltrzesniewski** [comment]
We can't get build triggers to work :/
This still needs some work, until then maybe you could try `<Prefer32Bit>false</Prefer32Bit>`

File: src/Abc.Zebus.Persistence.Runner/Program.cs [hunk 2]
```diff
@@ -38,32 +39,35 @@ public static void Main()
...
-            var busFactory = new BusFactory();
             var appSettingsConfiguration = new AppSettingsConfiguration();
-            InjectPersistenceServiceSpecificConfiguration(busFactory, appSettingsConfiguration);
-
-            busFactory
-                .WithConfiguration(appSettingsConfiguration, ConfigurationManager.AppSettings["Environment"])
-                .WithScan()
-                .WithEndpoint(ConfigurationManager.AppSettings["Endpoint"])
-                .WithPeerId(ConfigurationManager.AppSettings["PeerId"]);
+            var useCassandraStorage = ConfigurationManager.AppSettings["PersistenceStorage"] == "Cassandra";
```
[the remark this answers is not in the document]
Comment (ltrzesniewski) [reply src/Abc.Zebus.Persistence.Runner/Program.cs]: I don't really like the idea with running with a configuration that works by coincidence. ¯\\\_(ツ)\_/¯

**Comment by ocoanet** [comment]
I do not think that the acking mechanism is functional, especially because the `Acks` column family is never read and because the `RocksDbMessageReader` is ignoring acks.

I added a very simple replay test that fails.

Propositions:
1. Read the `Acks` column family on startup and pass the `_outOfOrderAcks` to the message reader.
2. Or change the acking design by removing the `Acks` column family and using a `Acked` column in the `Messages` column family.

**Comment by MendelMonteiro** [comment]
@ocoanet https://github.com/Abc-Arbitrage/Zebus/pull/78/commits/23425ef4f942ac15a796347fb6861f0270f4d90b addresses the clean up / initialisation of the out of order acks.
I also went ahead and changed the IMessageReader interface to return byte arrays instead of TransportMessages.

File: src/Abc.Zebus.Persistence.RocksDb/RocksDbStorage.cs [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+        public Task Write(IList<MatcherEntry> entriesToPersist)
+        {
+            foreach (var entry in entriesToPersist)
+            {
+                var key = CreateKeyBuffer(entry.PeerId);
+                FillKey(key, entry.PeerId, entry.MessageId.GetDateTime().Ticks, entry.MessageId.Value);
+                if (entry.IsAck)
+                {
+                    // Ack
+                    var bytes = _db.Get(key, _messagesColumnFamily);
```
Comment (ocoanet) [inline src/Abc.Zebus.Persistence.RocksDb/RocksDbStorage.cs]: This is a read for every ack. I feel like this is far from ideal for a high throughput usage considering RocksDB has a LSM storage. But I guess it should work for most usage cases and it is much simpler like that. I wish there was a `Exist` method in `RocksDb` though.

Other alternatives:
- Keep unacked message ids in-memory (not scalable).
- Use a write-only approach like in the Cassandra persistence (too complex).

File: src/Abc.Zebus.Persistence.RocksDb/RocksDbStorage.cs [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+            return nonAckedCounts;
+        }
+
+        public static void FillKey(byte[] key, PeerId peerId, long ticks, Guid messageId)
+        {
+            var peerPart = Encoding.UTF8.GetBytes(peerId.ToString());
+            Buffer.BlockCopy(peerPart, 0, key, 0, peerPart.Length);
+
+            var tickPart = BitConverter.GetBytes(ticks);
+            Array.Reverse(tickPart); // change endianness
```
Comment (ocoanet) [inline src/Abc.Zebus.Persistence.RocksDb/RocksDbStorage.cs]: I suppose `BitConverter.IsLittleEndian` could be used if we want to run this code on ARM :)

File: src/Abc.Zebus.Persistence.RocksDb/TransportMessageDeserializer.cs [hunk 5]
```diff
@@ -0,0 +1,15 @@
+﻿using Abc.Zebus.Serialization.Protobuf;
+using Abc.Zebus.Transport;
+
+namespace Abc.Zebus.Persistence.RocksDb
+{
+    internal static class TransportMessageDeserializer
+    {
+        public static TransportMessage Deserialize(byte[] bytes)
```
Comment (ocoanet) [inline src/Abc.Zebus.Persistence.RocksDb/TransportMessageDeserializer.cs]: This code is unused now.

**Pull request merged**